lib/internal/modules/esm/worker.js

```diff
@@ -0,0 +1,222 @@
+
+const lock = new Int32Array(workerData.lock);
+const { syncCommPort } = workerData;
+
+function transferArrayBuffer(hasError, source) {
+  if (hasError || source == null) return;
+  if (isArrayBuffer(source)) return [source];
+  if (isTypedArray(source)) return [TypedArrayPrototypeGetBuffer(source)];
+  if (isDataView(source)) return [DataViewPrototypeGetBuffer(source)];
+}
+
+function releaseLock() {
+  AtomicsNotify(lock, WORKER_TO_MAIN_THREAD_NOTIFICATION);
+}
+
+function errorHandler(err, origin = 'unhandledRejection') {
+  releaseLock();
+  process.off('uncaughtException', errorHandler);
+  if (hasUncaughtExceptionCaptureCallback()) {
+    process._fatalException(err);
+    return;
+  }
+  internalBinding('errors').triggerUncaughtException(
+    err,
+    origin === 'unhandledRejection',
+  );
+}
+
+/**
+ * ! Run everything possible within this function so errors get reported.
+ */
+(async function customizedModuleWorker() {
+  const { initializeESM, initializeHooks } = require('internal/modules/esm/utils');
+  let hooks;
+  let preloadScripts;
+  let hasInitializationError = false;
+  let initializationError;
+
+  try {
+    initializeESM();
+    const initResult = await initializeHooks();
+    hooks = initResult.hooks;
+    preloadScripts = initResult.preloadScripts;
+  } catch (exception) {
+    // If there was an error while parsing and executing a user loader, for
+    // example if because a loader contained a syntax error, then we need to
+    // send the error to the main thread so it can be thrown and printed.
+    hasInitializationError = true;
+    initializationError = exception;
+  }
+
+  syncCommPort.on('message', handleMessage);
+
+  if (hasInitializationError) {
+    syncCommPort.postMessage(wrapMessage('error', initializationError));
+  } else {
+    syncCommPort.postMessage(wrapMessage('success', { preloadScripts }), preloadScripts.map(({ port }) => port));
+  }
+
+  releaseLock(); // Send 'ready' signal to main.
+
+  let immediate;
+  let pendingAsyncResponses = 0;
+  function checkForMessages() {
+    immediate = setImmediate(checkForMessages).unref();
+    // We need to let the event loop tick a few times to give the main thread a
+    // chance to send follow-up messages.
+    const response = receiveMessageOnPort(syncCommPort);
+
+    if (response !== undefined) {
+      PromisePrototypeThen(handleMessage(response.message), undefined, errorHandler);
+    }
+  }
+
+  process.on('beforeExit', () => {
+    const value = AtomicsOr(lock, IDLENESS_OF_THREADS, WORKER_THREAD_IS_IDLE);
+
+    if (value & MAIN_THREAD_IS_IDLE && AtomicsLoad(lock, NUMBER_OF_INCOMING_MESSAGES) === 0) {
+      // Main thread is also out of work, there's nothing left to do.
+      return;
+    }
+
+    releaseLock(); // In case the main thread was waiting for a sync response.
+    // Let's pause the worker thread until a new message comes in.
```

Comment:
Are those 2 comment lines connected? If not:

```suggestion
    // Pause the worker thread until a new message comes in.
```

(the "Let's" feels like a connector/jointer)